data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -69,7 +69,8 @@ protected DeleteFilter(
       List<DeleteFile> deletes,
       Schema tableSchema,
       Schema requestedSchema,
-      DeleteCounter counter) {
+      DeleteCounter counter,
+      boolean isBatchReading) {
```

Comment:
Does batch reading necessarily mean no `_pos`? I believe not, as you can explicitly project it, like `select _pos from t`. We should give an accurate name, something like `needRowPosition` or `needRowPosCol`.

Reply:
Think a bit more, I'm considering if only the non-vector readers actually need an implicit `_pos` column. If that’s the case, would it make more sense to adjust this within RowReader by adding `_pos` there? This approach could simplify things by eliminating the need to check whether a reader is vectorized, especially since vectorization isn’t necessarily strongly correlated with the requirement for `_pos`.
Here is pseudo code to add it in class `RowDataReader`
```
    LOG.debug("Opening data file {}", filePath);
    expectedSchema().add(`_pos`);    <--- add it here
    SparkDeleteFilter deleteFilter =
        new SparkDeleteFilter(filePath, task.deletes(), counter(), false);
```